src/component/App.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react'
+import Display from './Display'
+import ButtonPanel from './ButtonPanel'
+import calculate from '../logic/calculate'
+import './App.css'
+import { AppState, clickHandlerFunction } from './types'
+
+// export default class App extends React.Component {
+//   state = {
+//     total: null,
+//     next: null,
+//     // eslint-disable-next-line react/no-unused-state
+//     operation: null,
+//   }
+
+//   handleClick = (buttonName) => {
+//     console.log(this.state)
+//     this.setState((prevState) => calculate(prevState, buttonName))
+//   }
+
+//   render() {
+//     return (
+//       <div className="component-app">
+//         <Display value={this.state.next || this.state.total || '0'} />
+//         <ButtonPanel clickHandler={this.handleClick} />
+//       </div>
+//     )
+//   }
+// }
+
+export default function App() {
+  const [ownState, setOwnState] = useState<AppState>({
+    total: null,
+    next: null,
+    operation: null,
+  })
+
+  const handleClick: clickHandlerFunction = (buttonName) => {
```

Comment:
The usage of the `Function` at the end of the function names is redundant.
```suggestion
  const handleClick: clickHandler= (buttonName) => {
```

Reply:
Yeah, it looks good without the Function part, but I think that the interfaces for Button's and ButtonPanel's Props need that to specify that the clickHandler is a Function 🤔 

```ts
export interface ButtonProps {
  name: string
  orange?: boolean
  wide?: boolean
  clickHandler: ClickHandlerFunction  // here
}

export interface ButtonPanelProps {
  clickHandler: ClickHandlerFunction  // here
}
```

Maybe I could create a type called `clickHandler` which derives from `clickHandlerFunction` and use it as the type for the handleClick function on `App.tsx`. What do you think?